### PR TITLE
PurchaseButtonComponent is now just a container/stack like ButtonComponent

### PR DESCRIPTION
**Pull request by joshdholtz**
### Motivation

Make `PurchaseButtonComponent` more flexible by just letting it be a stack (like `ButtonComponent`)

### Description

Removed all properties and just gave it a `stack: StackComponent` property

### CODE REVIEW

File: RevenueCatUI/Templates/Components/Button/ButtonComponentView.swift [hunk 1]
```diff
@@ -35,10 +35,12 @@ struct ButtonComponentView: View {
...
     var body: some View {
-        AsyncButton(
-            action: { try await performAction() },
-            label: { StackComponentView(viewModel: viewModel.stackViewModel, onDismiss: self.onDismiss) }
-        )
+        AsyncButton {
+            try await performAction()
+        } label: {
+            // Not passing an onDismiss - nothing in this stack should be able to dismiss
+            StackComponentView(viewModel: viewModel.stackViewModel, onDismiss: {})
```
Comment (JayShortway) [inline RevenueCatUI/Templates/Components/Button/ButtonComponentView.swift:42]: I don't think this is true? There's a dismiss button action (`navigateBack`).

**Review by JayShortway — COMMENTED** [review]
LGTM! 1 comment to double check.

File: RevenueCatUI/Templates/Components/Button/ButtonComponentView.swift (hunk 1 above)
Comment (joshdholtz) [reply RevenueCatUI/Templates/Components/Button/ButtonComponentView.swift]: Fixed!

**Review by JayShortway — APPROVED** [review]
Let's go!

**Pull request merged**